app/common/lib/ledgerUtil.js

```diff
@@ -524,7 +555,7 @@ const getMediaProvider = (url, firstPartyUrl, referrer) => {
       (firstPartyUrl && firstPartyUrl.startsWith('https://www.twitch.tv/')) ||
       (referrer && referrer.startsWith('https://player.twitch.tv/'))
     ) &&
-    url.startsWith('https://api.mixpanel.com/')
+    url.includes('ttvnw.net/v1/segment')
```

Comment:
this could lead to origin confusion, like if some site with a Twitch referrer embeds `https://evilsitettvnw.net/v1/segment` - can you make this check more specific?

Reply:
Not sure how I could do this. Any suggestions?

Reply:
either a regex or `url.includes('.ttvnw.net/v1/segment') || url.includes('https://ttvnw.net/v1/segment')` (if the top-level page is HTTPS, the resources are probably also HTTPS)`

Reply:
updated